pkg/operator/ceph/cluster/osd/health.go

```diff
@@ -146,3 +151,49 @@ func (m *Monitor) handleOSDMarkedOut(outOSDid int) error {
 	}
 	return nil
 }
+
+// restartOSDIfStuck will check if a portable OSD is on a node that is not ready.
+// If the pod is stuck in terminating state, go ahead and force delete the pod so K8s
+// will free up the volume and allow the OSD to be restarted on another node.
+func (m *Monitor) restartOSDIfStuck(osdID int) error {
+	labels := fmt.Sprintf("ceph-osd-id=%d,portable=true", osdID)
+	pods, err := m.context.Clientset.CoreV1().Pods(m.namespace).List(metav1.ListOptions{LabelSelector: labels})
```

Comment:
To make the loop a bit more efficient what about using `FieldSelector` with something like: 

`FieldSelector: fmt.Sprintf("status.phase!=%s", v1. PodReady)` 

So that the list returns only the failing pods and we don't have to iterate over all OSDs pods and check each status individually?

Reply:
We're only iterating over OSDs that are down and out, and it's only every 5 minutes. I'd still like to iterate over them to get the debug logging that indicates why a pod isn't stopped.

Reply:
Right, I overlooked that, sounds good to keep it this way then.